src/Workspaces/CSharp/Portable/Extensions/StatementSyntaxExtensions.cs

```diff
@@ -13,6 +13,17 @@ namespace Microsoft.CodeAnalysis.CSharp.Extensions
 {
     internal static class StatementSyntaxExtensions
     {
+        public static bool IsFirstStatementInEnclosingBlock(this StatementSyntax statement)
```

Comment:
note: the code may be enclosed in something that isn't a block.  for example, if you're in a `case:` clause in a `switch` statement.  please add a test for that.  thanks!